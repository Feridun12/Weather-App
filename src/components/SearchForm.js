import React from "react";

// eslint-disable-next-line react/prop-types
function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <input type="text" onChange={handleInputChange} value={searchText} />
      <button type="submit" onClick={onSubmit} data-testid="searchbutton">
        Search
      </button>
    </div>
  );
}

export default SearchForm;
