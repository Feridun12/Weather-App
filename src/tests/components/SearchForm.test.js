import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("renders and works correctly", () => {
  it("renders correctly (snapshot testing)", () => {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders elements correctly", () => {
    const { getByTestId } = render(<SearchForm />);

    expect(getByTestId("searchbutton")).toHaveTextContent("Search");
  });
});
