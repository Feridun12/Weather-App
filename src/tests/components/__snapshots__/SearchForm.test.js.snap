// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and works correctly renders correctly (snapshot testing) 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      type="text"
      value=""
    />
    <button
      data-testid="searchbutton"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
